CMakeLists.txt

```diff
@@ -167,19 +167,17 @@ if(HIOP_USE_GPU)
     set(CMAKE_CUDA_STANDARD_REQUIRED ON)
   endif()
 
-  if(HIOP_NVCC_ARCH)
-    set(CMAKE_CUDA_FLAGS "-arch=${HIOP_NVCC_ARCH}")
-    message(STATUS "Using CUDA arch ${HIOP_NVCC_ARCH}")
-  else()
-    set(CMAKE_CUDA_FLAGS "-arch=sm_35")
-    message(STATUS "Using CUDA arch sm_35")
+  if(NOT HIOP_NVCC_ARCH)
+    set(HIOP_NVCC_ARCH 35)
   endif()
-  set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --expt-extended-lambda")
-
+  set(CMAKE_CUDA_FLAGS "--expt-extended-lambda")
+  set(CMAKE_CUDA_ARCHITECTURES ${HIOP_NVCC_ARCH})
```

Comment:
Is `HIOP_NVCC_ARCH` cached anywhere? It would be good to allow user to change it.

Reply:
I pushed a commit that cache's this variable, along with overriding `CMAKE_CUDA_ARCHITECTURES`. We could have left this as-is and had the user override this CMake variable themselves, but it was a small change.

Reply:
I addressed this in the latest commit, and how this setting is stored in CMake cache